Get Unexpected line in ldd output:  /lib64/ld-linux-x86-64.so.2, after pull 102

Error info:
```
[$] ~/.local/bin/staticx --loglevel DEBUG dist/gdpy3-app/gdpy3-app \
    -l dist/gdpy3-app/ \
    -l /usr/lib/libm.so.6 -l /lib64/libpthread.so.0 \
    gdpy3-app
DEBUG:root:Running ['patchelf', '--set-interpreter', 'iiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiii', '--set-rpath', 'rrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrr', '--force-rpath', '/tmp/staticx-prog-zve_dwet']
INFO:root:Program interpreter: /lib64/ld-linux-x86-64.so.2
INFO:root:Using XZ BCJ filter FILTER_X86
INFO:root:Adding /tmp/staticx-prog-zve_dwet as .staticx.prog
DEBUG:root:Running ['ldd', 'dist/gdpy3-app/gdpy3-app']
staticx: Unexpected line in ldd output: 	/lib64/ld-linux-x86-64.so.2 => /usr/lib64/ld-linux-x86-64.so.2 (0x00007f79932aa000)
[$] ldd dist/gdpy3-app/gdpy3-app
	linux-vdso.so.1 (0x00007ffc431fc000)
	libdl.so.2 => /usr/lib/libdl.so.2 (0x00007f1de6361000)
	libz.so.1 => /usr/lib/libz.so.1 (0x00007f1de6347000)
	libc.so.6 => /usr/lib/libc.so.6 (0x00007f1de6180000)
	/lib64/ld-linux-x86-64.so.2 => /usr/lib64/ld-linux-x86-64.so.2 (0x00007f1de63ac000)
```
pull https://github.com/JonathonReinhart/staticx/pull/102, `staticx/elf.py` line 147-149

```
                if libpath:
                    raise LddError("Unexpected line in ldd output: " + line)
                yield libname
```

fix:
```
147     ¦   ¦   ¦   if libpath and not libpath.endswith('ld-linux-x86-64.so.2'):                                                        
148     ¦   ¦   ¦   ¦   raise LddError("Unexpected line in ldd output: " + line)
149     ¦   ¦   ¦   yield libname
```

I have tried your way of install. I get the same error.
```
pip install .
Processing /home/MyUsername/Downloads/staticx
Requirement already satisfied: pyelftools in /home/MyUsername/MyIDE/MyProject/venv/lib/python3.8/site-packages (from staticx==0.9.0+5.g7e84cd7) (0.26)
Installing collected packages: staticx
  Attempting uninstall: staticx
    Found existing installation: staticx 0.9.0
    Uninstalling staticx-0.9.0:
      Successfully uninstalled staticx-0.9.0
    Running setup.py install for staticx ... error
    ERROR: Command errored out with exit status 1:
     command: /home/MyUsername/MyIDE/MyProject/venv/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-req-build-v7f5n6dd/setup.py'"'"'; __file__='"'"'/tmp/pip-req-build-v7f5n6dd/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-xbplejbf/install-record.txt --single-version-externally-managed --compile --install-headers /home/MyUsername/MyIDE/MyProject/venv/include/site/python3.8/staticx
         cwd: /tmp/pip-req-build-v7f5n6dd/
    Complete output (84 lines):
    /home/MyUsername/MyIDE/MyProject/venv/lib/python3.8/site-packages/setuptools/dist.py:473: UserWarning: Normalizing '0.9.0+5-g7e84cd7' to '0.9.0+5.g7e84cd7'
      warnings.warn(
    running install
    running build
    running build_bootloader
    scons: Reading SConscript files ...
    scons: done reading SConscript files.
    scons: Building targets ...
    gcc -o scons_build/debug/bootloader/error.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -static -g -DDEBUG=1 -DSTATICX_VERSION=\"0.9.0+5-g7e84cd7\" -Ilibtar -Ilibxz bootloader/error.c
    gcc -o scons_build/debug/bootloader/elfutil.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -static -g -DDEBUG=1 -DSTATICX_VERSION=\"0.9.0+5-g7e84cd7\" -Ilibtar -Ilibxz bootloader/elfutil.c
    gcc -o scons_build/debug/bootloader/extract.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -static -g -DDEBUG=1 -DSTATICX_VERSION=\"0.9.0+5-g7e84cd7\" -Ilibtar -Ilibxz bootloader/extract.c
    gcc -o scons_build/debug/bootloader/main.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -static -g -DDEBUG=1 -DSTATICX_VERSION=\"0.9.0+5-g7e84cd7\" -Ilibtar -Ilibxz bootloader/main.c
    gcc -o scons_build/debug/bootloader/mmap.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -static -g -DDEBUG=1 -DSTATICX_VERSION=\"0.9.0+5-g7e84cd7\" -Ilibtar -Ilibxz bootloader/mmap.c
    gcc -o scons_build/debug/bootloader/util.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -static -g -DDEBUG=1 -DSTATICX_VERSION=\"0.9.0+5-g7e84cd7\" -Ilibtar -Ilibxz bootloader/util.c
    gcc -o scons_build/debug/libtar/block.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -g -DPACKAGE_VERSION=\"1.2.20\" -Ilibtar -Ilibxz -Iscons_build/debug/libtar/compat -Ilibtar/compat libtar/block.c
    gcc -o scons_build/debug/libtar/decode.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -g -DPACKAGE_VERSION=\"1.2.20\" -Ilibtar -Ilibxz -Iscons_build/debug/libtar/compat -Ilibtar/compat libtar/decode.c
    gcc -o scons_build/debug/libtar/extract.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -g -DPACKAGE_VERSION=\"1.2.20\" -Ilibtar -Ilibxz -Iscons_build/debug/libtar/compat -Ilibtar/compat libtar/extract.c
    gcc -o scons_build/debug/libtar/handle.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -g -DPACKAGE_VERSION=\"1.2.20\" -Ilibtar -Ilibxz -Iscons_build/debug/libtar/compat -Ilibtar/compat libtar/handle.c
    gcc -o scons_build/debug/libtar/output.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -g -DPACKAGE_VERSION=\"1.2.20\" -Ilibtar -Ilibxz -Iscons_build/debug/libtar/compat -Ilibtar/compat libtar/output.c
    gcc -o scons_build/debug/libtar/util.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -g -DPACKAGE_VERSION=\"1.2.20\" -Ilibtar -Ilibxz -Iscons_build/debug/libtar/compat -Ilibtar/compat libtar/util.c
    gcc -o scons_build/debug/libtar/compat/strlcpy.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -g -DPACKAGE_VERSION=\"1.2.20\" -Ilibtar -Ilibxz -Iscons_build/debug/libtar/compat -Ilibtar/compat libtar/compat/strlcpy.c
    gcc -o scons_build/debug/libtar/compat/strmode.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -g -DPACKAGE_VERSION=\"1.2.20\" -Ilibtar -Ilibxz -Iscons_build/debug/libtar/compat -Ilibtar/compat libtar/compat/strmode.c
    ar rc scons_build/debug/libtar/libtar.a scons_build/debug/libtar/block.o scons_build/debug/libtar/decode.o scons_build/debug/libtar/extract.o scons_build/debug/libtar/handle.o scons_build/debug/libtar/output.o scons_build/debug/libtar/util.o scons_build/debug/libtar/compat/strlcpy.o scons_build/debug/libtar/compat/strmode.o
    ranlib scons_build/debug/libtar/libtar.a
    Install file: "scons_build/debug/libtar/libtar.a" as "scons_build/debug/lib/libtar.a"
    gcc -o scons_build/debug/libxz/xz_crc32.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -g -DDEBUG=1 -DXZ_DEC_X86=1 -Ilibtar -Ilibxz libxz/xz_crc32.c
    gcc -o scons_build/debug/libxz/xz_dec_lzma2.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -g -DDEBUG=1 -DXZ_DEC_X86=1 -Ilibtar -Ilibxz libxz/xz_dec_lzma2.c
    gcc -o scons_build/debug/libxz/xz_dec_stream.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -g -DDEBUG=1 -DXZ_DEC_X86=1 -Ilibtar -Ilibxz libxz/xz_dec_stream.c
    gcc -o scons_build/debug/libxz/xz_dec_bcj.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -g -DDEBUG=1 -DXZ_DEC_X86=1 -Ilibtar -Ilibxz libxz/xz_dec_bcj.c
    ar rc scons_build/debug/libxz/libxz.a scons_build/debug/libxz/xz_crc32.o scons_build/debug/libxz/xz_dec_lzma2.o scons_build/debug/libxz/xz_dec_stream.o scons_build/debug/libxz/xz_dec_bcj.o
    ranlib scons_build/debug/libxz/libxz.a
    Install file: "scons_build/debug/libxz/libxz.a" as "scons_build/debug/lib/libxz.a"
    gcc -o scons_build/debug/bootloader/bootloader -static scons_build/debug/bootloader/error.o scons_build/debug/bootloader/elfutil.o scons_build/debug/bootloader/extract.o scons_build/debug/bootloader/main.o scons_build/debug/bootloader/mmap.o scons_build/debug/bootloader/util.o -Lscons_build/debug/lib -ltar -lxz
    /bin/ld: scons_build/debug/lib/libtar.a(output.o): in function `th_print_long_ls':
    /tmp/pip-req-build-v7f5n6dd/libtar/output.c:59: warning: Using 'getgrgid' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
    /bin/ld: scons_build/debug/lib/libtar.a(decode.o): in function `th_get_gid':
    /tmp/pip-req-build-v7f5n6dd/libtar/decode.c:65: warning: Using 'getgrnam' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
    /bin/ld: scons_build/debug/lib/libtar.a(decode.o): in function `th_get_uid':
    /tmp/pip-req-build-v7f5n6dd/libtar/decode.c:49: warning: Using 'getpwnam' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
    /bin/ld: scons_build/debug/lib/libtar.a(output.o): in function `th_print_long_ls':
    /tmp/pip-req-build-v7f5n6dd/libtar/output.c:52: warning: Using 'getpwuid' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
    gcc -o scons_build/release/bootloader/error.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -static -Os -DNDEBUG=1 -DSTATICX_VERSION=\"0.9.0+5-g7e84cd7\" -Ilibtar -Ilibxz bootloader/error.c
    gcc -o scons_build/release/bootloader/elfutil.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -static -Os -DNDEBUG=1 -DSTATICX_VERSION=\"0.9.0+5-g7e84cd7\" -Ilibtar -Ilibxz bootloader/elfutil.c
    gcc -o scons_build/release/bootloader/extract.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -static -Os -DNDEBUG=1 -DSTATICX_VERSION=\"0.9.0+5-g7e84cd7\" -Ilibtar -Ilibxz bootloader/extract.c
    gcc -o scons_build/release/bootloader/main.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -static -Os -DNDEBUG=1 -DSTATICX_VERSION=\"0.9.0+5-g7e84cd7\" -Ilibtar -Ilibxz bootloader/main.c
    In function 'create_tmpdir',
        inlined from 'main' at bootloader/main.c:393:20:
    bootloader/main.c:211:9: error: '%s' directive argument is null [-Werror=format-overflow=]
      211 |         error(2, errno, "Failed to create temporary directory: %s", tmpdir);
          |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    cc1: all warnings being treated as errors
    scons: *** [scons_build/release/bootloader/main.o] Error 1
    scons: building terminated because of errors.
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-req-build-v7f5n6dd/setup.py", line 46, in <module>
        setup(
      File "/home/MyUsername/MyIDE/MyProject/venv/lib/python3.8/site-packages/setuptools/__init__.py", line 145, in setup
        return distutils.core.setup(**attrs)
      File "/usr/lib/python3.8/distutils/core.py", line 148, in setup
        dist.run_commands()
      File "/usr/lib/python3.8/distutils/dist.py", line 966, in run_commands
        self.run_command(cmd)
      File "/usr/lib/python3.8/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
      File "/home/MyUsername/MyIDE/MyProject/venv/lib/python3.8/site-packages/setuptools/command/install.py", line 61, in run
        return orig.install.run(self)
      File "/usr/lib/python3.8/distutils/command/install.py", line 545, in run
        self.run_command('build')
      File "/usr/lib/python3.8/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/usr/lib/python3.8/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
      File "/tmp/pip-req-build-v7f5n6dd/setup.py", line 34, in run
        self.run_command('build_bootloader')
      File "/usr/lib/python3.8/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/usr/lib/python3.8/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
      File "/tmp/pip-req-build-v7f5n6dd/setup.py", line 29, in run
        check_call(args)
      File "/usr/lib/python3.8/subprocess.py", line 364, in check_call
        raise CalledProcessError(retcode, cmd)
    subprocess.CalledProcessError: Command '['scons']' returned non-zero exit status 2.
    ----------------------------------------
  Rolling back uninstall of staticx
  Moving to /home/MyUsername/MyIDE/MyProject/venv/bin/staticx
   from /tmp/pip-uninstall-2llzichm/staticx
  Moving to /home/MyUsername/MyIDE/MyProject/venv/lib/python3.8/site-packages/staticx-0.9.0.dist-info/
   from /home/MyUsername/MyIDE/MyProject/venv/lib/python3.8/site-packages/~taticx-0.9.0.dist-info
  Moving to /home/MyUsername/MyIDE/MyProject/venv/lib/python3.8/site-packages/staticx/
   from /home/MyUsername/MyIDE/MyProject/venv/lib/python3.8/site-packages/~taticx
ERROR: Command errored out with exit status 1: /home/MyUsername/MyIDE/MyProject/venv/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-req-build-v7f5n6dd/setup.py'"'"'; __file__='"'"'/tmp/pip-req-build-v7f5n6dd/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-xbplejbf/install-record.txt --single-version-externally-managed --compile --install-headers /home/MyUsername/MyIDE/MyProject/venv/include/site/python3.8/staticx Check the logs for full command output.

```
```
pip freeze
altgraph==0.17
cached-property==1.5.1
chardet==3.0.4
distro==1.4.0
patchelf-wrapper==1.1.0
pycrypto==2.6.1
pyelftools==0.26
PyInstaller==3.6
pylspci==0.3.1
pyusb==1.0.2
scons==3.1.2
staticx==0.9.0

```

--- a/staticx/elf.py
+++ b/staticx/elf.py
@@ -89,6 +89,8 @@ def _parse_ldd_output(output):
     #	linux-vdso.so.1 =>  (0x00007ffe53551000)
     #	libc.so.6 => /usr/lib64/libc.so.6 (0x00007f42ac010000)
     #	/lib64/ld-linux-x86-64.so.2 (0x0000557376e75000)
+    #     or
+    #   /lib64/ld-linux-x86-64.so.2 => /usr/lib64/ld-linux-x86-64.so.2 (0x00007f1de63ac000)
     pat = re.compile(r'\t([\w./+-]*) (?:=> ([\w./+-]*) )?\((0x[0-9a-fA-F]*)\)')
 
     for line in output.splitlines():
@@ -99,26 +101,28 @@ def _parse_ldd_output(output):
                 break
             raise LddError("Unexpected line in ldd output: " + line)
         libname  = m.group(1)
-        libpath  = m.group(2)
+        resolved = m.group(2)
         baseaddr = int(m.group(3), 16)
 
-        if libname.startswith('/'):
-            # An absolute path here is probably INTERP
-            # and ldd shouldn't output the => /abs/path part.
-            if libpath:
-                raise LddError("Unexpected line in ldd output: " + line)
-            yield libname
+        if resolved:
+            # ldd outupt a resolved symlink, use that
+            libpath = resolved
+        elif libname.startswith('/'):
+            # The library directly referenced an absolute path, use that
+            libpath = libname
         else:
-            # A short libname should come from a NEEDED tag
-            # and ldd should include the => /abs/path part.
-            # If it doesn't, then it's probably linux-vdso*.so
-            # or linux-gate.so
-            if not libpath:
-                # TODO: This check could be removed/relaxed
-                if libname.startswith('linux-'):
-                    continue
-                raise LddError("Unexpected line in ldd output: " + line)
-            yield libpath
+            # Assume an unresolved non-absolute library name is synthetic
+            # like linux-vdso*.so or linux-gate.so which should be ignored
+            # TODO: This check could be removed/relaxed
+            if not libname.startswith('linux-'):
+                raise LddError("Unexpected unresolved libname: " + libname)
+            logging.debug("Ignoring synthetic library: " + libname)
+            continue
+
+        # The library path must be absolute
+        if not libpath.startswith('/'):
+            raise LddError("Unexpected non-absolute libpath: " + libpath)
+        yield libpath
 
 
 def get_shobj_deps(path, libpath=[]):

--- a/unittest/test_elf.py
+++ b/unittest/test_elf.py
@@ -39,3 +39,20 @@ class TestLdd:
             "/lib64/ld-linux-x86-64.so.2",
         ]
         self._test(output, exp)
+
+
+    def test_arch(self):
+        # https://github.com/JonathonReinhart/staticx/issues/116
+        output = (
+            "\tlinux-vdso.so.1 (0x00007ffc431fc000)\n"
+            "\tlibdl.so.2 => /usr/lib/libdl.so.2 (0x00007f1de6361000)\n"
+	    "\tlibz.so.1 => /usr/lib/libz.so.1 (0x00007f1de6347000)\n"
+	    "\tlibc.so.6 => /usr/lib/libc.so.6 (0x00007f1de6180000)\n"
+	    "\t/lib64/ld-linux-x86-64.so.2 => /usr/lib64/ld-linux-x86-64.so.2 (0x00007f1de63ac000)\n")
+        exp = [
+            "/usr/lib/libdl.so.2",
+	    "/usr/lib/libz.so.1",
+	    "/usr/lib/libc.so.6",
+	    "/usr/lib64/ld-linux-x86-64.so.2",
+        ]
+        self._test(output, exp)

